plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/ConnectionWrapper.java

```diff
@@ -204,9 +214,11 @@ private void updateWithResources(List<IProject> newValue) {
 			projects.clear();
 			for (IProject r : newValue) {
 				ProjectWrapper existingWrapper = oldWrappers.remove(r);
-
 				if (existingWrapper == null) {
 					ProjectWrapper newWrapper = new ProjectWrapper(this, r);
+					resourceCache.add(r);

```

Comment:
This seems like the wrong place to begin the watch.  My thought is we would start watch on connection expansion, maybe this is the only place to insert it.  Also, I would not expect to need to provide the project name for a watch; watches provide callbacks based on resource kind and you receive callbacks for all resource of that kind for which you have access.